app/deliveryoptions/utils.py

```diff
@@ -0,0 +1,727 @@
+    surrogate_data: List,
+) -> str:
+    return html_builder(
+        delivery_option_data, record_data, surrogate_data=surrogate_data
+    )
+
+
+# Specific pre-processing for the heading
+def heading_builder(
+    delivery_option_data: str, record_data: Dict, surrogate_data: List
+) -> str:
+    return html_builder(
+        delivery_option_data, record_data, surrogate_data=surrogate_data
+    )
+
+
+def surrogate_link_builder(surrogates: List) -> Tuple[List[Any], List[Any]]:
+    surrogate_list = []
+    av_media_list = []
+
+    for s in surrogates:
+        if s["xReferenceURL"]:
+            surrogate_list.append(s["xReferenceURL"])
+
+        if s["xReferenceType"] == "AV_MEDIA":
+            if s["xReferenceURL"]:
+                av_media_list.append(s["xReferenceURL"])
+
+    return surrogate_list, av_media_list
+
+
+def is_onsite(visitor_ip_address) -> bool:
+    return is_ip_in_cidr(visitor_ip_address, IP_ONSITE_RANGES)
+
+
+def is_subscribed() -> bool:
+    # TODO once user management is in place
+    return False
+
+
+def is_staff(visitor_ip_address) -> bool:
+    return is_ip_in_cidr(visitor_ip_address, IP_STAFFIN_RANGES)
+
+
+def get_dev_reader_type() -> Reader:
+    override_reader_type = os.getenv("OVERRIDE_READER_TYPE", Reader.UNDEFINED)
+
+    # If environment variable is set, validate it
+    if override_reader_type != Reader.UNDEFINED:
+        try:
+            # Convert the environment variable to an integer
+            reader_value = int(override_reader_type)
+
+            # Check if it's a valid Reader enum value
+            if reader_value in Reader._value2member_map_:
+                return Reader(
+                    reader_value
+                )  # Return the corresponding Reader enum value
+
+        except Exception as e:
+            logger.warning(
+                f"Override reader type '{override_reader_type}' cannot be determined - returning UNDEFINED ({type(e)}: {e.args})"
+            )
+
+    return Reader.UNDEFINED  # Default to UNDEFINED
+
+
+def get_reader_type(request: HttpRequest) -> Reader:
+    reader = Reader.UNDEFINED
+
+    try:
+        visitor_ip_address = get_client_ip(request)
+    except Exception as e:
+        logger.warning(
+            f"Cannot determine the users ip address - returning OFFSITE ({type(e)}: {e.args})"
+        )
+        return Reader.OFFSITE
+
+    # Check if there is an override of reader type - used for testing and demonstrations.
+    reader = get_dev_reader_type()
+
+    if reader == Reader.UNDEFINED:
+        if is_subscribed():
+            reader = Reader.SUBSCRIPTION
+        elif is_onsite(visitor_ip_address):
+            reader = Reader.ONSITEPUBLIC
+        elif is_staff(visitor_ip_address):
+            reader = Reader.STAFFIN
+        else:
+            reader = Reader.OFFSITE
+
+    return reader
+
+
+# Main routine called from records.py
+def construct_delivery_options(
+    doptions: list, record: dict, request: HttpRequest
+) -> dict:
```

Comment:
```suggestion
def construct_delivery_options(
    delivery_options: list, record: Record, request: HttpRequest
) -> dict:
```
`delivery_options` - keep same name as passed parameter, 
`record`: is a type of `Record` -> from app.records.models import Record

Reply:
Agreed and changed